### ISSUE TITLE
In twisted/spread, fix use of list and range in Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8619]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8619|
|Type|enhancement|
|Created|2016-07-18 07:25:55Z|
|Branch|https://github.com/twisted/twisted/tree/8619-rodrigc-spread|

There are a few places in twisted/spread which use range()
that return errors under Python 3, because in Python 3
range() returns an iterator, not a list.

<details><summary>Searchable metadata</summary>

```
trac-id__8619 8619
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8619_rodrigc_spread 8619-rodrigc-spread
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468826755665309 1468826755665309
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/348

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset dd8df60e30c752d62d2c86b25be104928a78e408
```
#!CommitTicketReference repository="" revision="dd8df60e30c752d62d2c86b25be104928a78e408"
Merge pull request #348 from twisted/8619-rodrigc-spread

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8619

In twisted/spread, fix use of list and range
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
